pipeline/compute_metrics.py

```diff
@@ -28,94 +28,221 @@
+                    raise RuntimeError("Some data is missing!")
+                data.append(evaluator.fill_up_missing())
                 continue
-            if is_correct_key not in line_dict:
-                fill_up_missing(evaluations, log_probs, pred_answers, allow_incomplete)
+            if evaluator.is_incomplete(line_dict):
+                if not allow_incomplete:
+                    raise RuntimeError("Some data is missing!")
+                data.append(evaluator.fill_up_missing())
                 continue
-            pred_answers.append(line_dict.get("predicted_answer"))
-            evaluations.append(line_dict[is_correct_key])
-            log_probs.append(line_dict.get("log_prob"))
-            if line_dict.get("log_prob") is None and aggregation_mode == "most_probable":
-                raise RuntimeError(
-                    "Cannot compute most probable generation because some of the probabilities are missing"
-                )
-
-        total += 1
-        if aggregation_mode == "best":
-            total_correct += any(evaluations)
-            if all([ans is None for ans in pred_answers]):
-                total_no_answer += 1
-        elif aggregation_mode == "majority":
-            # TODO: currently majority does not take into account equivalent answers written in a different way
-            valid_answers_and_results = [
-                (ans, is_correct) for ans, is_correct in zip(pred_answers, evaluations) if ans is not None
-            ]
-            if len(valid_answers_and_results) == 0:
-                total_no_answer += 1
-            else:
-                majority_result = Counter(valid_answers_and_results).most_common(1)[0][0]
-                total_correct += majority_result[1]
-        elif aggregation_mode == "first":
-            total_correct += evaluations[0]
-            total_no_answer += pred_answers[0] is None
-        elif aggregation_mode == "most_probable":
-            most_probable_result = sorted(zip(log_probs, evaluations, pred_answers))[-1]
-            total_correct += most_probable_result[1]
-            total_no_answer += most_probable_result[2] is None
-        else:
-            raise ValueError(f"Unsupported mode {aggregation_mode}")
+            data.append(line_dict)
+
+        evaluator.update(data, aggregation_mode)
 
     for file_handle in file_handles:
         file_handle.close()
 
-    if eval_type == "math":
-        correct_answer = total_correct / total * 100.0
-        no_answer = total_no_answer / total * 100.0
-        wrong_answer = (total - total_correct - total_no_answer) / total * 100.0
-    elif eval_type == "code":
-        correct_answer = total_correct / total * 100.0
-        no_answer = 0
-        wrong_answer = (total - total_correct) / total
-    else:
-        raise ValueError(f"Unsupported eval_type {eval_type}")
-    return correct_answer, wrong_answer, no_answer, total
+    return evaluator.get_metrics()
+
+
+math_benchmarks = [
```

Comment:
Would be good to move these benchmark-related constants to a separate file

Reply:
Yeah, eventually, but maybe not in this PR?

Reply:
Sure, thing. May be we can add a ToDo